nemo/core/config/set_config.py

```diff
@@ -25,6 +25,57 @@
 from nemo.core.config import Config
 
 
+def hydra_runner(
+    config_path: Optional[str] = None, config_name: Optional[str] = None
+) -> Callable[[TaskFunction], Any]:
+    """
+    Decorator used for passing the Config paths to main function.
+
+    Args:
+        config_path: path to the directory where the config exists.
+        config_name: name of the config file.
+    """
+
+    def decorator(task_function: TaskFunction) -> Callable[[], None]:
+        @functools.wraps(task_function)
+        def wrapper(cfg_passthrough: Optional[DictConfig] = None) -> Any:
+            # Check it config was passed.
+            if cfg_passthrough is not None:
+                return task_function(cfg_passthrough)
+            else:
+                args = get_args_parser()
+
+                # Parse arguments in order to retrieve overrides
+                parsed_args = args.parse_args()
+
+                # Get overriding args in dot string format
+                overrides = parsed_args.overrides  # type: list
+
+                # Update overrides
+                overrides.append("hydra.run.dir=.")
+                overrides.append('hydra.job_logging.root.handlers=null')
```

Comment:
@blisc do I understand it correctly that this is required:
 - only for DDP
 - when DDP, config saving works properly for global rank = 0, but messes up for the others?

I generally liked the idea of automatic saving of the whole configuration that hydra did... is there any chance that we could leave it for no global rank/global rank=0?

Reply:
Hydra changing the current_directory really messes up a lot of things. Yes hydra works neatly for single-gpu, but it fails for multi-gpu. As it stands, files are still logged to "./.hydra", but not all the ranks will try to write to that file at the same time.
I am trying to think what I can do to get around that, and the only answer I can come up with is to write a decorator that does `os.cwd()` right before the script exits.

Reply:
Ok, let's merge this PR as is and then I will wrestle with this a bit. Makes sense?